### ISSUE TITLE
Crash on win32_start_service with php.exe

### ISSUE THREAD

**Issue by macintoshplus** [InExtenso/win32service#6]
1- add `'path' => '"' . dirname(PHP_BINARY) . '\\php-win.exe"',` at line 15 into `sample.php`
2- run `php sample.exe install`
3- run `php service_dummy.php`

Crash stack:

```
>	php.exe!main(int argc, char * * argv) Ligne 1358	C
 	[Code externe]	
```

PHP version:

```
C:\php\debug>php -v
PHP 7.1.0 (cli) (built: Jan  5 2017 08:55:43) ( NTS MSVC14 (Visual C++ 2015) x64 DEBUG )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.1.0-dev, Copyright (c) 1998-2016 Zend Technologies
```

PHP build command: `configure.bat --with-mp=auto --enable-cli --disable-zts --enable-win32service=shared --with-php-build=deps --enable-debug --enable-mbstring --enable-cli-win32`

**Comment by macintoshplus** [comment]
No crash on branch PHP-7.0.

**Comment by alphp** [comment]
In my tests tonight I used my branch https://github.com/alphp/win32service/tree/no-cli-constraint that only has one modification in the code to remove the load restriction of the extension in CLI SAPI.

I have tested both with php-7.0.14-ts-x64 and php-7.1.0-ts-x64 with correct results with php-win as binary for main service loop and php.exe for all other commands.

**Comment by macintoshplus** [comment]
The SAPI name for `php-win.exe` is `cli`. The removed code have no effect.

In PHP 7.1 this code has been added into the `php_cli.c` file 
```
#if defined(PHP_WIN32) && !defined(PHP_CLI_WIN32_NO_CONSOLE)
	php_win32_cp_cli_setup();
	orig_cp = (php_win32_cp_get_orig())->id;
	/* Ignore the delivered argv and argc, read from W API. This place
		might be too late though, but this is the earliest place ATW
		we can access the internal charset information from PHP. */
	argv_wide = CommandLineToArgvW(GetCommandLineW(), &num_args);
	PHP_WIN32_CP_W_TO_ANY_ARRAY(argv_wide, num_args, argv, argc)
	using_wide_argv = 1;

	SetConsoleCtrlHandler(php_cli_win32_ctrl_handler, TRUE);
#endif
```

This code attempt to get the current console codepage only for `php.exe`. But in the service context, is not a console context.

I agree that it is necessary to use `php-win.exe` but `php.exe` does not have to crash in this context.

I have open an bug on PHP bug tracker available here: https://bugs.php.net/bug.php?id=73870

**Comment by alphp** [comment]
I want to use win32service with apache, so I have removed the CLI SAPI restriction.

**Comment by macintoshplus** [comment]
Ok, for apache or IIS, I prefer use specific configuration file. If you want remove this restriction or change the method to restrict usage, please open an new issue.

**Comment by alphp** [comment]
OK, maybe when we have something more stable.
For now having my own fork and unofficial compilation is enough.
Later on I can make a push request with the proposed change.

**Comment by macintoshplus** [comment]
This extension work with PHP 7.1.1-dev but not with PHP 7.1.0.

**Comment by alphp** [comment]
Now that you say it, you've already noticed.
It does not work if you compile against PHP-7.0.0 or PHP-7.1.0.
If you compile against 7.0.x or 7.1.x then it works.

**Comment by macintoshplus** [comment]
Fixed with PR #10
